Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2368,11 +2332,7 @@ ExceptionOr<Ref<ImageData>> CanvasRenderingContext2DBase::getImageData(int sx, i
 
     ASSERT(pixelBuffer->format().colorSpace == toDestinationColorSpace(computedColorSpace));
 
-    if (canvasBase().postProcessPixelBuffer(*pixelBuffer, m_wasLastDrawPutImageData, suppliedColors())) {
-        IntSize destOffset { 0, 0 };
-        IntRect sourceRect = computeImageDataRect(*buffer, imageDataRect.width(), imageDataRect.height(), imageDataRect, destOffset);
-        buffer->putPixelBuffer(*pixelBuffer, sourceRect, IntPoint { destOffset });
-    }
+    canvasBase().postProcessPixelBufferResults(*pixelBuffer, suppliedColors(), imageDataRect);
```

Comment:
This call seems like a layering violation, still.
I'd consider it a better approach if above, you had:
```
canvasBase().makeRenderingResultsAvailable();
ImageBuffer* buffer = canvasBase().buffer();
````

In other words: the post processing would be always maintained before an escaping read.
So this would be doing the post processing for

Reply:
That's a good suggestion. The one reason I am using `postProcessPixelBufferResults` here is because this is a case where the "post processing" is not written back to the ImageBuffer and it does not result in repainting the canvas with the changes. Ideally, post-processing should only affect the script-readable extracted data. Currently, this isn't true because we rely on `makeRenderingResultsAvailable` for handling the other extraction methods. I see your point that this is a layering violation, and maybe this isn't sufficiently valuable.

Reply:
> Ideally, post-processing should only affect the script-readable extracted data.

I think this is a really good idea, but the code has to realize this, then, ideally for all cases. I think mix-and-match sometimes mutating sometimes not is a bad combination.